docs/cluster_spec.md

```diff
@@ -190,14 +190,44 @@ More information about running in an existing VPC is [here](run_in_existing_vpc.
 
 ### hooks
 
-Hooks allow the execution of a container before the installation of Kubneretes on every node in a cluster.  For intance you can install nvidia drivers for using GPUs.
+Hooks allow for the execution of an action before the installation of Kubernetes on every node in a cluster.  For instance you can install nvidia drivers for using GPUs. This hooks can be in the form on docker images or manifest files (systemd units). Hooks can be place in either then cluster spec, meaning the will be globally deployed, or they can be placed into the instanceGroup specification. Note, service names on the instanceGroup which overlap with the cluster spec take precedence and ignore the cluster spec definition, i.e. if you have a unit file 'myunit.service' in cluster and then one in the instanceGroup, only the instanceGroup is applied.
 
 ```
 spec:
   # many sections removed
   hooks:
-  - execContainer:
+  - before:
+    - some_service.service
+    requires:
+    - docker.service
+    execContainer:
       image: kopeio/nvidia-bootstrap:1.6
+
+  # or a raw systemd unit
+  hooks:
+  - name: iptable-restore.service
+    masterOnly: true|false # only run this on masters
```

Comment:
Although I've now seen that you have hooks on the instancegroup level also (which is great), so I'm thinking we could just have a list of roles.  And as we don't have that many roles, this might not much matter...